platform/server/plugins/types.ts

```diff
@@ -77,6 +77,17 @@ export interface BasePluginsType {
   [key: string]: any;
 }
 
+export type KibanaPlugin<
```

Comment:
Naming feels a bit confusing now, we have `KibanaPlugin`, `KibanaFunctionalPlugin`, `KibanaClassPlugin` and `KibanaPluginStatic`. Maybe it's just me, but If I look only at the names then I have an impression that `KibanaPlugin` and `KibanaFunctionalPlugin` represent the same "type" of things. But in reality `KibanaPlugin` is more like plugin definition object while `KibanaFunctionalPlugin` is a "real" plugin (or plugin register/start/init function).

What do you think?

Reply:
++ agreed. Made some changes, does it look better now?